### ISSUE TITLE
version 0.22.0 broke discovery of configuration in package.json

### ISSUE THREAD

**Issue by dmccaffery** [DavidAnson/markdownlint-cli2#818]
Given the following markdown:

```markdown
# Sample

This line is intentionally far longer than the default markdownlint MD013 line-length limit of eighty characters so that the rule will fire whenever the configuration in package.json is not picked up.
```

with the following configuration in `package.json`:

```json
{
  "name": "markdownlint-ignores-package-json",
  "version": "1.0.0",
  "private": true,
  "description": "Reproduction: markdownlint-cli2 >= 0.22.0 ignores configuration in package.json",
  "markdownlint-cli2": {
    "config": {
      "default": true,
      "MD013": false
    }
  }
}
```

markdownlint-cli2@0.21.0 correctly picks up the configuration and ignores the line-length while markdownlint-cli2@0.22.x does not pick up the configuration and continues reporting an error.

For a full reproduction:

https://github.com/dmccaffery/markdownlint-ignores-package-json

**Comment by DavidAnson** [comment]
Use of package.json is no longer automatic, but it can still be used (more flexibly!) with --config and --configPointer. The docs explain how and include an example: https://github.com/DavidAnson/markdownlint-cli2#command-line

**Comment by dmccaffery** [comment]
A note to others that use package.json for config and come across this issue:

This breaks existing integrations such as nvim-lint:

https://github.com/mfussenegger/nvim-lint/blob/d48f3a76189d03b2239f6df1b2f7e3fa8353743b/lua/lint/linters/markdownlint-cli2.lua#L5

While it is possible to use a workaround whereby the arguments are customised in neovim configs, this would apply to every repo, and would be error-prone as we'd have to implement our own discovery logic to "do the right thing" depending on the repo. Its possible, but would duplicate a lot of logic that already exists.

I think a better workaround is to stop using package.json configuration entirely and use a separate config file.

**Comment by DavidAnson** [comment]
@dmccaffery Sorry for the trouble! Here's a bit of context: Python people asked for similar special treatment for their ecosystem version of package.json which was already a special case file because it only applies at the root. I did not want to add another magic file, so I decided to generalize current support to work with both - or anything similar. Having done that, it seemed I could remove the special case for package.json without loss of generality. I didn't realize it would affect baked-in assumptions like you refer to, but also this change rolled out about two months ago and this issue is the first I think anyone brought this up. If it is causing absolute havoc, I'm open to brainstorming, but my stance is that I would prefer to keep things as they are now.

**Comment by dmccaffery** [comment]
Oh, I'm not arguing with you -- it is your project and I appreciate that it exists. I left the above note there for others that may find this issue in the future when they (inevitably) run into this at some point. 

I'm happy to implement the "workaround". It's not a lot of trouble to pull the config out into its own file. The only reason I integrate it into package.json is in situations where the package.json is there for other reasons; it's just a little nicer to keep things tidy / in one place in the ecosystem.

I apologise if it came across as though I was challenging the decision. Since it hasn't reached 1.0 yet, I can't even complain about the BREAKING CHANGE. :)
